CONTRIBUTING.md

```diff
@@ -0,0 +1,193 @@
+# Programming Guidelines
+
+## General Guidelines
+
+### Why do we even need guidelines in the first place?
+
+At the beginning, Space Concordia Robotics was small. The code base was maintained by two individuals that knew the code very well. The robotics team is now growing with new members of a wide range of programming experience. Due to the negligence of the best practices, the code became spaghetti and almost impossible to read for new members. It is hard to debug and to test. We need clean code because
+
+#### Clean code is easier to read
+
+#### Clean code is easier to debug and is more maintainable
+
+#### Well designed code is scalable
+
+#### Well planned code is fun to build on
+
+#### There is a cost to having a total mess
+Teams that have moved swiftly at the beginning of projects can find themselves moving at a much slower rate further in. The term [Technical Debt](https://en.wikipedia.org/wiki/Technical_debt) describes the cost of the additional work that needs to be done (because of slower debugging, more bugs, etc) due to messy code.
+
+## Best Practices
+
+These best practices are mostly taken from the book `Clean Code, A Handbook of Agile Craftsmanship` by Robert C. Martin 
+
+### 1. Functions
+
+#### 1.1 Do one thing
+
+Functions should do one thing. They should do it well. They should do it only. One symptom of doing more than one thing is having sections within functions. These can oftentimes easily be split up into subroutines that will only have a specific role.
```

Comment:
"They should do it alone" sounds bit better to me here than "They should do it only" but I think either works to get the point across.

Reply:
Additionally, 1.1 only applies to functions in this document it could be extended to classes as well as per the [swiss army knife anti-pattern](https://sourcemaking.com/antipatterns/swiss-army-knife)

Reply:
> "They should do it alone" sounds bit better to me here than "They should do it only" but I think either works to get the point across.

It's actually a quote from a software engineering book on clean code that I copy pasted

Reply:
Good point with this antipattern

Reply:
is this gucci now?

Reply:
gucci